web/annotation_layer_builder.js

```diff
@@ -257,6 +259,8 @@ class AnnotationLayerBuilder {
     if (!this.#externalHide) {
       this.div.hidden = false;
     }
+
+    this.#linksInjected = true;
```

Comment:
The same line is should also be added to the early-return branch above https://github.com/mozilla/pdf.js/blob/e3cca6d513d4c24393cf5d8d4af25d69b399b46e/web/annotation_layer_builder.js#L246-L248 since otherwise there'd still be unnecessary re-parsing happening (even if nothing renders in that case).

Reply:
Since the only other code path would be to go through with the entire rest of the function, what would be wrong with just moving the statement above this branch?

Reply:
Right, it could just be placed directly after https://github.com/mozilla/pdf.js/pull/19470/files#diff-b166eae766e0c30a0a7dba9e750bd020d5ce9b432b84d6cb722aab3a8ca95aa1R242 instead.